api/bag_private.h

```diff
@@ -1,192 +1,113 @@
-bagError bagWriteAttribute  (bagHandle hnd, hid_t lid, u8 *attr_name, void *value);
-bagError bagReadAttribute   (bagHandle hnd, hid_t lid, u8 *attr_name, void *value);
-bagError bagAlignAttribute  (bagHandle hnd, hid_t lid, u8 *attr_name, void *value, u32 read_or_write);
-bagError bagReadSurfaceDims (bagHandle hnd, hsize_t *max_dims);
-bagError bagReadOptSurfaceDims (bagHandle hnd, s32 type);
-bagError bagAlignXMLStream  (bagHandle hnd, s32 read_or_write);
-bagError bagAlignRow        (bagHandle hnd, u32 row, u32 start_col,u32 end_col, s32 type, s32 read_or_write, void *data);
-bagError bagAlignRegion     (bagHandle hnd, u32 start_row, u32 start_col, u32 end_row, u32 end_col, s32 type, s32 read_or_write, hid_t xfer);
-bagError bagAlignNode       (bagHandle hnd, u32 row, u32 col, s32 type, void *data, s32 read_or_write);
-bagError bagAlignOptRow     (bagHandle hnd, u32 row, u32 start_col,u32 end_col, s32 type, s32 read_or_write, void *data);
-bagError bagAlignOptRegion  (bagHandle hnd, u32 start_row, u32 start_col, u32 end_row, u32 end_col, s32 type, s32 read_or_write, hid_t xfer);
-bagError bagAlignOptNode    (bagHandle hnd, u32 row, u32 col, s32 type, void *data, s32 read_or_write);
-bagError bagUpdateMinMax    (bagHandle hnd, u32 type);
-bagError bagReadTrackingList(bagHandle hnd, u16 mode, u32 inp1, u32 inp2, bagTrackingItem **items, u32 *rtn_len);
-bagError bagFillPos         (bagHandle hnd, u32 r1, u32 c1 , u32 r2, u32 c2, f64 **x, f64 **y);
-bagError bagSortTrackingList(bagHandle hnd, u16 mode);
-s32 bagCompareTrackIndices  (const void *a, const void *b);
-s32 bagCompareTrackNodes    (const void *a, const void *b);
-s32 bagCompareTrackCodes    (const void *a, const void *b);
-
-#endif
+namespace BAG
+{
+
+//! The maximum compression level supported by HDF5.
+constexpr int kMaxCompressionLevel = 9;
+
+//! Path names for BAG entities
+#define ROOT_PATH                       "/BAG_root"
+#define METADATA_PATH                   ROOT_PATH "/metadata"
+#define ELEVATION_PATH                  ROOT_PATH "/elevation"
+#define UNCERTAINTY_PATH                ROOT_PATH "/uncertainty"
+#define TRACKING_LIST_PATH              ROOT_PATH "/tracking_list"
+
+//! Path names for optional BAG entities
+#define NOMINAL_ELEVATION_PATH          ROOT_PATH "/nominal_elevation"
+#define VERT_DATUM_CORR_PATH            ROOT_PATH "/vertical_datum_corrections"
+#define HYPOTHESIS_STRENGTH_PATH        ROOT_PATH "/hypotheses_strength"
+#define NUM_HYPOTHESES_PATH             ROOT_PATH "/num_hypotheses"
+#define AVERAGE_PATH			        ROOT_PATH "/average"
+#define NODE_GROUP_PATH			        ROOT_PATH "/node"
+#define ELEVATION_SOLUTION_GROUP_PATH   ROOT_PATH "/elevation_solution"
+#define SHOAL_ELEVATION_PATH            ROOT_PATH "/shoal_elevation"
+#define STANDARD_DEV_PATH	            ROOT_PATH "/standard_dev"
+#define NUM_SOUNDINGS_PATH              ROOT_PATH "/num_soundings"
+#define COMPOUND_PATH                   ROOT_PATH "/Georef_metadata/"
```

Comment:
I find confusing that the "Georef_metadata" is called "COMPOUND_PATH" here. Similarly, the "bag_compound_layer" should be renamed "bag_georef_metadata".

Reply:
I agree with changing it if it make the code more readable.

Reply:
This has been addressed.